client/parser/gen.go

```diff
@@ -0,0 +1,391 @@
+package parser
+
+import (
+	"alda.io/client/model"
+	"fmt"
+)
+
+// appendDuration appends a DurationNode child for a non-empty model.Duration.
+func appendDuration(node ASTNode, duration model.Duration) (ASTNode, error) {
+	if len(duration.Components) == 0 {
+		return node, nil
+	}
+
+	durationNode := ASTNode{Type: DurationNode}
+
+	for _, component := range duration.Components {
+		switch c := component.(type) {
+
+		case model.Barline:
+			durationNode.Children = append(durationNode.Children, ASTNode{
+				Type: BarlineNode,
+			})
+
+		case model.NoteLength:
+			noteLength := ASTNode{Type: NoteLengthNode, Children: []ASTNode{{
+				Type:    DenominatorNode,
+				Literal: c.Denominator,
+			}}}
+			if c.Dots > 0 {
+				noteLength.Children = append(noteLength.Children, ASTNode{
+					Type:    DotsNode,
+					Literal: c.Dots,
+				})
+			}
+			durationNode.Children = append(durationNode.Children, noteLength)
+
+		case model.NoteLengthMs:
+			durationNode.Children = append(durationNode.Children, ASTNode{
+				Type:    NoteLengthMsNode,
+				Literal: c.Quantity,
+			})
+
+		default:
+			// model.NoteLengthBeats - only generated within lisp
+			return ASTNode{}, fmt.Errorf(
+				"unexpected DurationComponent during AST generation: %#v", c,
+			)
+
+		}
+	}
+
+	node.Children = append(node.Children, durationNode)
+	return node, nil
+}
+
+// mapIsolatedUpdate maps a single isolated model.ScoreUpdate to ASTNode.
+// Holistic updates that require "re-construction" are handled upstream:
+// 	1. model.PartDeclaration in GenerateASTFromScoreUpdates.
+// 	2. model.VoiceMarker, model.VoiceGroupEndMarker in mapUpdatesWithoutParts.
+func mapIsolatedUpdate(updates model.ScoreUpdate) (ASTNode, error) {
```

Comment:
`updates` is a confusing name for a single ScoreUpdate. Maybe this should be called `update`, and `update` below could just be `u`?